pages/previewModel.js

```diff
@@ -86,11 +115,20 @@ function loadModel(gltf, resetCamera) {
 
         viewer._model = model;
         viewer.onLoad.raiseEvent(model);
-    }).otherwise(function(e){
+    }).otherwise(function(e) {
         window.alert('Error: ' + e);
     });
 }
 
-var gltf = JSON.parse(document.getElementById('glTF').textContent);
+var gltfFileName = document.getElementById('gltfFileName').textContent;
+var gltfRootPath = document.getElementById('gltfRootPath').textContent;
 
-loadModel(gltf, true);
+try {
+    var gltfContent = JSON.parse(document.getElementById('gltf').textContent);
+    loadModelFromContent(gltfContent, gltfRootPath, true);
+}
+catch (ex) {
+    // If the glTF content is missing or not valid JSON, then try to load the
+    // model directly from the glTF file.
+    loadModelFromFile(gltfFileName, gltfRootPath, true);
```

Comment:
Actually -- that may not be completely true.  I added `"*"` as one of the activation events in package.json` per the [docs](https://code.visualstudio.com/docs/extensionAPI/activation-events) and now it's _partially_ working.  Within the context of the preview window, `GltfPreviewDocumentContentProvider.provideTextDocumentContent` gets invoked any time any protocol is invoked that isn't natively supported (so, `gltf-preview://`, `gltf-dataUri://` and even `foo://bar`).  The problem is, the incoming `uri` is always identical to the initial `uri` that initiated the preview request.  The actual `uri` that spawned the second request from the preview window just somehow gets dropped.  I'm debugging further to see if this is by design or a bug within VS Code.